### ISSUE TITLE
Avatars not showing when not using HTTPS

### ISSUE THREAD

**Issue by guusdk** [conversejs/converse.js#2374]
The rendering of avatars from vCards uses `crypto.subtle`. This feature is available only in secure contexts (HTTPS), in some or all supporting browsers ([source](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle)).

This results in avatars not being shown when the Converse webapp is opened on a non-secure context.

![Screenshot from 2020-12-24 17-18-50](https://user-images.githubusercontent.com/4253898/103098725-2210a480-460c-11eb-984d-5502ffe582ba.png)
![Screenshot from 2020-12-24 17-18-35](https://user-images.githubusercontent.com/4253898/103098718-1e7d1d80-460c-11eb-8cab-beeda7216cb8.png)

**Comment by Echolon** [comment]
rel. https://github.com/conversejs/converse.js/issues/1130

**Comment by kmxunan** [comment]
Hi @jcbrand 👋

I noticed this avatar issue too — the `crypto.subtle.digest('SHA-1', ...)` API requires a secure context (HTTPS) and silently fails on HTTP.

**Proposed approach:**
1. Wrap `crypto.subtle` calls in try/catch in the 3 affected files (vcard parsers)
2. Fall back to a lightweight JS-based SHA-1 implementation (e.g., using the Web Crypto API polyfill pattern) when `crypto.subtle` is unavailable
3. Add tests verifying avatar hashing works in non-secure contexts

Would you add the **bounty** label to this issue as well? Happy to prioritize based on which gets labeled first.

**Comment by jcbrand** [comment]
@kmxunan I agree with your suggested approach, but I don't think this issue is worthy of a bounty.

It's in my opinion not important enough.

For now, if people use Converse purely locally, they can use localhost, which allows `crypto.subtle` and if they host via a webserver they should use TLS.
